In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os
%matplotlib inline

data = pd.DataFrame()
data['Name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Oliver'] 
data['Age'] = [14, 12, 11, 10, 8, 6, 8]
print(data)

     Name  Age
0    Greg   14
1  Marcia   12
2   Peter   11
3     Jan   10
4   Bobby    8
5   Cindy    6
6  Oliver    8


In [7]:
data.describe()

Age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [11]:
np.std(data['Age'] ,ddof=1) / np.sqrt(len(data['Age']))

1.0334540197243192

For central tendancy, I would just pick the mean because of the relatively small standard deviation.

For variance, I would choose the Standard Deviation bceause we have the entire population we are evaluating and have no uncertaintity in the estimate of the mean.

In [15]:
data_update = data
data_update['Age'] = [14, 12, 11, 10, 8, 7, 8]
data_update.describe()

Age
count   7.000000
mean   10.000000
std     2.516611
min     7.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

The only changes that occured when Cindy turned 7 was the change in std, mean and min. Quartiles, max, and counts all remained the same.

In [16]:
data_update['Age'] = [14, 12, 11, 10, 8, 7, 1]
data_update['Name'] = ['Greg', 'Marcia', 'Peter', 'Jan', 'Bobby', 'Cindy', 'Jessica']

data_update.describe()

Age
count   7.000000
mean    9.000000
std     4.242641
min     1.000000
25%     7.500000
50%    10.000000
75%    11.500000
max    14.000000

You would use the standard deviation still, but it would be more fruitful to use the medin now (50%) as the estimte of cnetral tendancy because of the 1 year old outlier that severley increased the Standard Deviation.

In [19]:
fans = pd.DataFrame() 
fans['Percent'] = [20, 23, 17, 5]
fans.describe()

Percent
count   4.000000
mean   16.250000
std     7.889867
min     5.000000
25%    14.000000
50%    18.500000
75%    20.750000
max    23.000000

The lowest entry was for SciPhi Phanatic which (if closely related to ScienceFiction) would not be a proper estimate of the entire population, yet a specific sample of adults that varied from the population. Because of this outlier, it would be best to take the Median of the population and roughly estimate 18.5% percent of adult Americans were Brady Bunch fans on the 50th anniversay of the show.